src/core/src/HttpAssetAccessor.cpp

```diff
@@ -109,7 +112,11 @@ CesiumAsync::Future<std::shared_ptr<CesiumAsync::IAssetRequest>> HttpAssetAccess
     session->SetHeader(cprHeader);
     session->SetUrl(cpr::Url(url));
     session->GetCallback([promise, url, headers](cpr::Response&& response) mutable {
-        promise.resolve(std::make_shared<HttpAssetRequest>("GET", url, headers, std::move(response)));
+        if (response.error) {
+            promise.reject(std::runtime_error("failed to fetch tileset from: " + url + "\n"));
```

Comment:
Style thoughts:

* Capitalize the error message
* Use fmt for string formatting rather than string concatenation
* Is the newline needed?

Also, it might make sense to include the response error in the message.

Reply:
Also, this error can get printed for any types of requests, not just tilesets, so the message should be more generic, like what cesium-unity is doing: https://github.com/CesiumGS/cesium-unity/blob/82c3be4f7cfd0b3f120e57b75771f5d3b46c222d/native~/Shared/src/UnityAssetAccessor.cpp#L187-L188

Reply:
Thanks! Incorporated into latest commit